src/bindings/python/src/pyopenvino/core/common.hpp

```diff
@@ -80,22 +80,99 @@ std::vector<size_t> get_strides(const py::array& array);
 
 py::array as_contiguous(py::array& array, ov::element::Type type);
 
+template <typename T>
+py::array array_from_tensor_t(ov::Tensor&& t, py::dtype&& dtype) {
+    std::vector<T> tmp;
+    if (t.get_element_type() == ov::element::bf16) {
+        for (size_t i = 0; i < t.get_size(); i++) {
+            tmp.emplace_back(static_cast<T>(ov::bfloat16::from_bits(*(reinterpret_cast<uint16_t*>(t.data()) + i)))); 
+        }
+    } else {
+        tmp = ov::op::v0::Constant(t).cast_vector<T>();
+    }
+    return py::array(dtype, t.get_shape(), tmp.data());
+}
+
 py::array array_from_tensor(ov::Tensor&& t, bool is_shared);
 
+py::array array_from_tensor(ov::Tensor&& t, const ov::element::Type& dst_dtype);
+
+py::array array_from_constant(ov::op::v0::Constant&& c, bool is_shared);
+
+template <typename T>
+std::vector<T> array_as_vector(py::array& array){
+    T *ptr = static_cast<T*>(const_cast<void*>(array.data()));
+    return std::vector<T>(ptr, ptr + array.size());
+}
+
+template <typename T>
+std::vector<char> array_as_vector_bool(py::array& array) {
+    std::vector<char> result;
+    result.reserve(array.size());
+
+    for(long int i = 0; i < array.size(); i++) {
+        result.emplace_back(*(static_cast<T*>(const_cast<void*>(array.data())) + i) != 0 ? 1 : 0);
```

Comment:
Why const cast is required?
is array.data() pointing to `bool` type.
Is possible to change this loop to something similar like:
```cpp
std::transform(array.begin(), array.end(), std::back_inserter(result), [](const bool v){ return static_cast<char>(v);});
```
cast bool value will return 0 or 1.